pkg/routeagent_driver/handlers/ovn/non_gateway_route_handler.go

```diff
@@ -142,6 +146,16 @@ func (h *NonGatewayRouteHandler) TransitionToGateway() error {
 		return nil
 	}
 
+	for _, endpoint := range h.remoteEndpoints {
+		// This piece of code is designed to manage upgrades from a version lower than 0.16.3 to a higher version,
+		// where we utilize the endpoint name as the identifier for ngwr. It can be removed once we stop supporting
+		// the 0.16 version.
+		if err := h.smClient.SubmarinerV1().NonGatewayRoutes(endpoint.Namespace).Delete(context.TODO(),
+			endpoint.Name, metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
+			return errors.Wrapf(err, "error deleting nonGatewayRoute %q", endpoint.Name)
+		}
```

Comment:
This can be combined with the loop below on L159.

Reply:
Done.